inc/zoo/swar/SWAR.h

```diff
@@ -77,6 +77,21 @@ struct SWAR {
 
     constexpr T value() const noexcept { return m_v; }
 
+    template<std::size_t N>
+    constexpr static T baseFromLaneLiterals(const T(&args)[N]) {
+        static_assert(N == Lanes, "Wrong number of lanes");
```

Comment:
SFINAE this.
`std::enable_if_t<N == Lanes, T>`
Also, please separate the type from the identifier being declared as in `const T (&args)[`, in this codebase we bind the pointer and reference indicators to the identifier they apply to, not to the type they don't apply to, as is far more common.

Reply:
(y)